openeo_processes_dask/process_implementations/ml/curve_fitting.py

```diff
@@ -0,0 +1,118 @@
+from typing import Callable, Optional
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from numpy.typing import ArrayLike
+
+from openeo_processes_dask.process_implementations.cubes import apply_dimension
+from openeo_processes_dask.process_implementations.data_model import RasterCube
+from openeo_processes_dask.process_implementations.exceptions import (
+    DimensionNotAvailable,
+)
+
+__all__ = ["fit_curve", "predict_curve"]
+
+
+def fit_curve(data: RasterCube, parameters: list, function: Callable, dimension: str):
+    if dimension not in data.dims:
+        raise DimensionNotAvailable(
+            f"Provided dimension ({dimension}) not found in data.dims: {data.dims}"
+        )
+
+    dims_before = list(data.dims)
+
+    # In the spec, parameters is a list, but xr.curvefit requires names for them,
+    # so we do this to generate names locally
+    parameters = {f"param_{i}": v for i, v in enumerate(parameters)}
+
+    # The dimension along which to predict cannot be chunked!
+    rechunked_data = data.chunk({dimension: -1})
```

Comment:
Shouldn't we do the same in `predict_curve` as well?

Reply:
I don't think it matters for predict, because there each timestep can be inferenced for independently of the other steps!

Reply:
@LukeWeidenwalker that's true. But I was wondering if it would be faster when predicting on a datacube with many timesteps?

Reply:
Hmm - not sure tbh, I haven't profiled `predict_curve` at all yet - I think I'll merge and deploy this now so we can start a training run at least, and revisit this if performance of inference turns out to be a problem!